### PR TITLE
Enhance default-message-card.tmpl to prevent json parse errors

### PR DESCRIPTION
**Pull request by AnandDevan**
Log errors from json.Compact call for usability

### CODE REVIEW

**Comment by AnandDevan** [comment]
This change is able to handle newline, single quote etc in the received payload. Example:
```   {
      "status": "firing",
      "labels": {
        "account_id": "xxx",
        "alertname": "EndpointDown",
        "environment": "dev",
        "instance": "xxxx",
        "job": "blackbox_URL_Monitor",
        "pod": "xxx",
        "region": "us-west-2",
        "severity": "Low"
      },
      "annotations": {
        "description": "'Endpoint \t \n **'ADDDDDD'** xxxxxx",
        "summary": "Endpoint xxxxx"
      },
      "startsAt": "2019-12-03T15:13:20.729714813Z",
      "endsAt": "0001-01-01T00:00:00Z",
      "generatorURL": "xxxxxxxxx"
  }
```

**Comment by Knappek** [comment]
@AnandDevan thank you for this PR, fiiinally I was able to go through it. Looks good, just one small comment.

File: default-message-card.tmpl [hunk 1]
```diff
@@ -10,29 +10,29 @@
...
   {{- range $index, $alert := .Alerts }}{{- if $index }},{{- end }}
     {
-      "activityTitle": "[{{ $alert.Annotations.description }}]({{ $externalUrl }})",
+      "activityTitle": "[{{ js $alert.Annotations.description |  reReplaceAll "_" "\\\\_" | reReplaceAll `\\'` "'" }}]({{ $externalUrl }})",
       "facts": [
         {{- range $key, $value := $alert.Annotations }}
         {
-          "name": "{{ reReplaceAll "_" "\\\\_" $key }}",
-          "value": "{{ reReplaceAll "_" "\\\\_" $value }}"
+          "name": "{{ $key }}",
```
[the remark this answers is not in the document]
Comment (AnandDevan) [reply default-message-card.tmpl]: I found that the fact name does not  honor markup syntax.
With this payload (note the first fact name with underscores):
```
{
  "summary": "Card \"Test card\"",
  "themeColor": "0078D7",
  "title": "Card created: \"Name of card\"",
  "sections": [
    {
      "activityTitle": "David Claux",
      "activitySubtitle": "9/13/2016, 3:34pm",
      "activityImage": "https://connectorsdemo.azurewebsites.net/images/MSC12_Oscar_002.jpg",
      "facts": [
        {
          "name": "this_is_Board:",
          "value": "Name_of_board"
        },
        {
          "name": "List:",
          "value": "Name of list"
        },
        {
          "name": "Assigned to:",
          "value": "(none)"
        },
        {
          "name": "Due date:",
          "value": "(none)"
        }
      ],
      "text": "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat."
    }
  ]
}
```

The message looks this this in teams:
image

[the remark this answers is not in the document]
Comment (Knappek) [reply default-message-card.tmpl]: ha you are right, didn't notive that.

**Pull request merged**